modules/ocf_ldap/files/generate_gh_known_hosts

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env python3
+import ipaddress
+import textwrap
+
+import requests
+
+GITHUB_META = 'https://api.github.com/meta'
+GITHUB_PUBLIC_KEY = 'AAAAB3NzaC1yc2EAAAABIwAAAQEAq2A7hRGmdnm9tUDbO9IDSwBK6TbQa+PXYPCPy6rbTrTtw7PHkccKrpp0yVhp5HdEIcKr6pLlVDBfOLX9QUsyCOV0wzfjIJNlGEYsdlLJizHhbn2mUjvSAHQqZETYP81eFzLQNnPHt4EVVUh7VfDESU84KezmD5QlWpXLmvU31/yMf+Se8xhHTvKSCZIFImWwoG6mbUoWf9nzpIoaSjB+weqqUUmpaaasXVal72J+UX2B+2RPW3RcT0eOzQgqlJL3RKrTJvdsjE3JEAvGq3lGHSZXy28G3skua2SmVi/w4yCE6gbODqnTWlg7+wC604ydGXA8VJiS5ap43JXiUFFAaQ=='  # noqa: E501
+
+
+def fetch_github_ip_range(key='git'):
+    """Fetch a list of IPs from github's meta API endpoint"""
+    response = requests.get(GITHUB_META)
+    response.raise_for_status()
+    return response.json()[key]
+
+
+def subnet_to_wildcard(subnet):
+    """Turn an IP network object into a wildcard form appropriate for a SSH
+    known_hosts file.
+    """
+    prefix = subnet.prefixlen
+
+    # The subnet given must be reduced down to a "class" so that we can make a
+    # wildcard out of it. If this is not the case then it needs to be split up
+    # further.
+    assert prefix % 8 == 0, 'Subnet must be reduced to a class (multiple of 8) first'
+
+    net_addr = str(subnet.network_address)
+    # Short-circuit if we get a /32 since it's just an IP address and doesn't
+    # need a wildcard added anywhere
+    if prefix == 32:
+        return net_addr
+
+    octets = net_addr.split('.')
+    octets_to_keep = prefix // 8
+    return '.'.join(octets[:octets_to_keep] + ['*'])
+
+
+def ip_range_to_wildcard(cidr_range):
+    """Translate an IP range (in CIDR notation) to a wildcard form appropriate
+    for a SSH known_hosts file.
+
+    This only works for IPv4 addresses currently since that's all that GitHub supports.
```

Comment:
If GitHub ever becomes IPv6 compatible, hopefully someone notices the rootspam this causes and fixes it :)